indico/MaKaC/webinterface/pages/admins.py

```diff
@@ -70,80 +72,25 @@ def _getHeader( self ):
-        userManagementSection = wcomponents.SideMenuSection(_('User management'), id='user_management',
-                                                            icon='icon-users')
-        pluginsSection = wcomponents.SideMenuSection(_('Plugins'), id='plugins', icon='icon-settings')
-        customizationSection = wcomponents.SideMenuSection(_('Customization'), id='customization', icon='icon-wrench')
-        integrationSection = wcomponents.SideMenuSection(_('Integration'), id='integration', icon='icon-earth')
-
-        self._sideMenu.addSection(mainSection)
-        self._sideMenu.addSection(securitySection)
-        self._sideMenu.addSection(userManagementSection)
-        self._sideMenu.addSection(pluginsSection)
-        self._sideMenu.addSection(customizationSection)
-        self._sideMenu.addSection(integrationSection)
-
-        self._generalSettingsMenuItem = wcomponents.SideMenuItem(
-            _("General settings"),
-            urlHandlers.UHAdminArea.getURL(), section='general')
-        self._sideMenu.addItem(self._generalSettingsMenuItem)
-
-        self._domainsMenuItem = wcomponents.SideMenuItem(
-            _("IP Domains"),
-            urlHandlers.UHDomains.getURL(), section='security')
-        self._sideMenu.addItem(self._domainsMenuItem)
-
-        self._templatesMenuItem = wcomponents.SideMenuItem(
-            _("Layout"),
-            urlHandlers.UHAdminLayoutGeneral.getURL(), section='customization')
-        self._sideMenu.addItem(self._templatesMenuItem)
-
-        self._ipProtectionMenuItem = wcomponents.SideMenuItem(
-            _("IP-based ACL"),
-            urlHandlers.UHIPBasedACL.getURL(), section='security')
-        self._sideMenu.addItem(self._ipProtectionMenuItem)
-
-        self._homepageMenuItem = wcomponents.SideMenuItem(
-            _("Homepage"),
-            urlHandlers.UHUpdateNews.getURL(), section='customization')
-        self._sideMenu.addItem(self._homepageMenuItem)
-
-        self._systemMenuItem = wcomponents.SideMenuItem(
-            _("System"),
-            urlHandlers.UHAdminsSystem.getURL(), section='general')
-        self._sideMenu.addItem(self._systemMenuItem)
-
-        self._protectionMenuItem = wcomponents.SideMenuItem(
-            _("Protection"),
-            urlHandlers.UHAdminsProtection.getURL(), section='security')
-        self._sideMenu.addItem(self._protectionMenuItem)
-
-        self.extra_menu_items = {}
-        for name, item in sorted(values_from_signal(signals.admin_sidemenu.send()),
-                                 key=lambda x: x[1]._title):
-            self.extra_menu_items[name] = item
-            self._sideMenu.addItem(item)
-
-    def _getBody( self, params ):
-        self._createSideMenu()
-        self._setActiveSideMenuItem()
+    def _getAdminSideMenu(self):
+        menu = menu_from_signals('admin-sidemenu',
+                                 active_item=self._get_active_menu_item())
```

Comment:
Why not use `self.sidemenu_option` directly? If it has to be dynamic you can always override the class-level attribute with a property (since you always access it via `self` - and a classproperty would probably work, too)